### ISSUE TITLE
What should be done about translatable strings after the sim split?

### ISSUE THREAD

**Issue by AgustinVallejo** [phetsims/my-solar-system#87]
My Solar System (MSS) was split into MSS, Solar System Common (SSC) and Kepler's Laws (KpL). However, the strings file in MSS had already been translated, so it's not that easy to split as well. Right now, SSC has no strings, KpL is using the strings from MSS, which is not ideal, and I think some string keys have also changed, which could cause some translation issues...

@marlitas suggested we use a script made by @chrisklus to address this problem. We're currently looking into it.

@zepumph also volunteered to help out, thanks!

**Comment by zepumph** [comment]
After discussing this with @chrisklus, since he just did this with number-suite sims, I have a path forward and I'm working on it now.

Step one, fix the strings in master of the repos, and hold off on babel.
Step two, likely to wait until something is republished, update babel and move translations to the right repo.

VERY IMPORTANT: please make sure that no one renames a string key from here, as it will very much complicate the process (@AgustinVallejo responsible dev).

**Comment by zepumph** [comment]
With further help from @chrisklus, I have fixed all en strings on master, moving them to their appropriate repo. Strings used by both my-solar-system AND keplers-laws ended up in solar-system-common.

We found these unused string keys. I removed them in the commit. Most likely they can be forgotten about UNLESS they were previously translated, and there is another key that is basically a "rename" that we want to keep these translations for. @AgustinVallejo is that true for any of these?

`orbital`
`centerOrbit`
`values`
`area.title`
`area.sweptArea`
`area.areaGraph`
`semiMajorAxis`
`period`
`symbols.eccentricity`
`units.units`
`units.Mjup`
`pattern.textValueUnits`
`pattern.rangeWithUnits`
`pattern.rangeWithExponentialUnits`

We also checked and have no more unused strings, and no string keys are duplicated across repos (using a custom version of `rosetta/scripts/js/remove-babel-duplicate-strings`)

**Comment by zepumph** [comment]
Ok. @chrisklus and I got things to a good place. So now we only have steps once the new version of my-solar-system is published. It is to run these scripts to make sure that translations that live in my-solar-system are appropriately moved to the other repos

```bash
cd babel
../rosetta/bin/copy-babel-subdirectory.sh my-solar-system keplers-laws
../rosetta/bin/copy-babel-subdirectory.sh my-solar-system solar-system-common
cd ..
node rosetta/scripts/js/remove-translations-without-english-key.js keplers-laws
node rosetta/scripts/js/remove-translations-without-english-key.js solar-system-common 
```

A few notes about running this:
* At the time of this work, there were no keys from existing translations whatsoever that were moved into the keplers-laws repo. If at the end of running these commands, there are only empty objects in the keplers laws files, then likely you should not commit them!
* I would recommend not running this as one big batch, but instead run each line individually. @chrisklus and I remain optimistic that this is the complete work to be done upon my-solar-system publication, but anything could happen!

**Comment by zepumph** [comment]
Over to @AgustinVallejo to review the unused keys for atypical "renames" and commits in general.

**Comment by AgustinVallejo** [comment]
Thanks @zepumph and @chrisklus!! Great work

**Comment by zepumph** [comment]
Today @chrisklus and I edited https://github.com/phetsims/my-solar-system/issues/87#issuecomment-1449041208 again to be a bit more clear (upon looking at https://github.com/phetsims/number-suite-common/issues/27).

The steps in that comment should happen BEFORE the final production deploy of MSS, and not after publication. Updating labels accordingly.

**Comment by zepumph** [comment]
>We found these unused string keys. I removed them in the commit. Most likely they can be forgotten about UNLESS they were previously translated, and there is another key that is basically a "rename" that we want to keep these translations for. @AgustinVallejo is that true for any of these?
> `orbital`
> `centerOrbit`
> `values`
> `area.title`
> `area.sweptArea`
> `area.areaGraph`
> `semiMajorAxis`
> `period`
> `symbols.eccentricity`
> `units.units`
> `units.Mjup`
> `pattern.textValueUnits`
> `pattern.rangeWithUnits`
> `pattern.rangeWithExponentialUnits`

`values` was the only one published with the 1.0 prototype, and there is no need to keep translations of it.

>Step 2: Double check again that there are no overlapping or problematic string keys. I did this during this work, but a double check before continuing will be helpful.

@AgustinVallejo and I wrote a script that showed that path is duplicated in my-solar-system and solar-system-common:

<details>

```
const mySolarStrings = {
  'my-solar-system.title': {
    'value': 'My Solar System'
  },
  'screen.intro': {
    'value': 'Intro'
  },
  'screen.lab': {
    'value': 'Lab'
  },
  'centerOfMass': {
    'value': 'Center of Mass'
  },
  'followCenterOfMass': {
    'value': 'Follow Center of Mass'
  },
  'returnBodies': {
    'value': 'Return Bodies'
  },
  'path': {
    'value': 'Path'
  },
  'dataPanel.X': {
    'value': 'x'
  },
  'dataPanel.Y': {
    'value': 'y'
  },
  'dataPanel.Vx': {
    'value': 'V<sub>x</sub>'
  },
  'dataPanel.Vy': {
    'value': 'V<sub>y</sub>'
  },
  'mass': {
    'value': 'Mass'
  },
  'dataPanel.mass': {
    'value': 'Mass'
  },
  'dataPanel.velocity': {
    'value': 'Velocity'
  },
  'dataPanel.position': {
    'value': 'Position'
  },
  'dataPanel.bodies': {
    'value': 'Bodies'
  },
  'dataPanel.moreData': {
    'value': 'More Data'
  },
  'mode.sunAndPlanet': {
    'value': 'Sun, Planet'
  },
  'mode.sunPlanetAndMoon': {
    'value': 'Sun, Planet, Moon'
  },
  'mode.sunPlanetAndComet': {
    'value': 'Sun, Planet, Comet'
  },
  'mode.trojanAsteroids': {
    'value': 'Trojan Asteroids'
  },
  'mode.ellipses': {
    'value': 'Ellipses'
  },
  'mode.hyperbolic': {
    'value': 'Hyperbolic'
  },
  'mode.slingshot': {
    'value': 'Slingshot'
  },
  'mode.doubleSlingshot': {
    'value': 'Double Slingshot'
  },
  'mode.binaryStarPlanet': {
    'value': 'Binary Star, Planet'
  },
  'mode.fourStarBallet': {
    'value': 'Four Star Ballet'
  },
  'mode.doubleDouble': {
    'value': 'Double Double'
  },
  'mode.custom': {
    'value': 'Custom'
  },
  'units.kg': {
    'value': '10<sup>28</sup> kg'
  },
  'unitsInfo.title': {
    'value': 'Units Information'
  },
  'unitsInfo.content': {
    'value': '<b>Note 1:</b> An Astronomical Unit (AU) is about 150 million kilometers. This is the average distance between the Sun and the Earth.'
  },
  'unitsInfo.content2': {
    'value': '<b>Note 2:</b> The mass values in this simulation range from 1 Jupiter mass to 1.5 Solar masses. All astronomical objects show a similar behavior under a strong gravitational field, including smaller planets, comets, and asteroids.'
  },
  'unitsInfo.content3': {
    'value': '<b>Note 3:</b> Due to the large range of values in the simulation, the gravity force vectors may sometimes be too small or too large to display. The gravity force vectors can be rescaled with the zoom control.'
  },
  'pattern.labelParenthesesUnits': {
    'value': '{{label}} ({{units}})'
  },
  'pattern.range': {
    'value': '{{min}} to {{max}}'
  },
  'a11y': {
    'moreData': {
      'value': 'More Data'
    },
    'info': {
      'value': 'Info'
    },
    'numberOfBodies': {
      'value': 'Number of Bodies'
    },
    'simDescription': {
      'value': '{{My Solar System}} is an interactive sim. It changes as you play with it. It has a Play Area and a Control Area.'
    },
    'introScreen': {
      'screenSummary': {
        'playAreaDescription': {
          'value': 'The Play Area is an observation window showing a scene with two planetary bodies. Experiment with the mass, velocity, location and direction of the two planetary bodies.'
        },
        'controlAreaDescription': {
          'value': 'In the Control Area, there are options to add a grid and a measuring tape. You can also toggle on or off the path and values of your planetary bodies and also check their center of mass. The Control Area has options for how fast the action happens in the observation window including buttons to pause and step forward.'
        }
      }
    },
    'labScreen': {
      'screenSummary': {
        'playAreaDescription': {
          'value': 'The Play Area is an observation window showing a scene with many options to set up your own experiments. Recreate our own Sun and Earth or make up your own solar system.'
        },
        'controlAreaDescription': {
          'value': 'In the Control Area, there are options to view and change the area of the orbit and options to adjust and view orbital information. You can choose premade planetary alignments or choose your own, along with a Zoom in and out option. The Control Area has options for how fast the action happens in the observation window including buttons to pause and step forward.'
        }
      },
      'modeSelector': {
        'value': 'Mode Selector'
      }
    }
  }
};

const keplersStrings = {
  'keplers-laws.title': {
    'value': 'Kepler\'s Laws'
  },
  'screen.firstLaw': {
    'value': 'First Law'
  },
  'screen.secondLaw': {
    'value': 'Second Law'
  },
  'screen.thirdLaw': {
    'value': 'Third Law'
  },
  'screen.lab': {
    'value': 'Lab'
  },
  'undefined': {
    'value': 'undefined'
  },
  'circularOrbit': {
    'value': 'Always circular'
  },
  'axis': {
    'value': 'Axes'
  },
  'foci': {
    'value': 'Foci'
  },
  'strings': {
    'value': 'String'
  },
  'semiaxis': {
    'value': 'Semiaxes'
  },
  'eccentricity': {
    'value': 'Eccentricity'
  },
  'eccentricityEquation': {
    'value': 'Eccentricity ='
  },
  'apoapsis': {
    'value': 'Apoapsis'
  },
  'periapsis': {
    'value': 'Periapsis'
  },
  'starMass': {
    'value': 'Star Mass'
  },
  'ourSun': {
    'value': 'Our Sun'
  },
  'period': {
    'value': 'Period'
  },
  'area.periodDivision': {
    'value': 'Period Divisions'
  },
  'area.values': {
    'value': 'Area Values'
  },
  'area.areaUnits': {
    'value': 'Area (AU<sup>2</sup>)'
  },
  'sweptArea': {
    'value': 'Swept Area'
  },
  'graph.title': {
    'value': 'a{{axisPower}} vs T{{periodPower}} Graph'
  },
  'graph.a': {
    'value': 'Semimajor Axis (a)'
  },
  'graph.t': {
    'value': 'Period (T)'
  },
  'symbols.semiMajorAxis': {
    'value': 'a'
  },
  'symbols.semiMinorAxis': {
    'value': 'b'
  },
  'symbols.focalDistance': {
    'value': 'c'
  },
  'symbols.period': {
    'value': 'T'
  },
  'symbols.radius': {
    'value': 'r'
  },
  'symbols.distance1': {
    'value': 'd<sub>1</sub>'
  },
  'symbols.distance2': {
    'value': 'd<sub>2</sub>'
  },
  'pattern.textEqualsValueUnits': {
    'value': '{{text}} = {{value}} {{units}}'
  },
  'warning.warning': {
    'value': 'Warning'
  },
  'warning.crashOrbit': {
    'value': 'The body will crash into the sun'
  },
  'warning.escapeOrbit': {
    'value': 'The body will escape the system'
  }, a11y: {}
};

const commonStrings = {
  'gravityForce': {
    'value': 'Gravity Force'
  },
  'grid': {
    'value': 'Grid'
  },
  'measuringTape': {
    'value': 'Measuring Tape'
  },
  'velocity': {
    'value': 'Velocity'
  },
  'V': {
    'value': 'V'
  },
  'clear': {
    'value': 'Clear'
  },
  'path': {
    'value': 'Path'
  },
  'zoom': {
    'value': 'Zoom:'
  },
  'offscaleMessage': {
    'value': 'Some force vectors are too small to display.'
  },
  'speed': {
    'value': 'Speed (km/s)'
  },
  'units.AU': {
    'value': 'AU'
  },
  'units.years': {
    'value': 'years'
  },
  'units.kms': {
    'value': 'km/s'
  },
  'pattern.labelUnits': {
    'value': '{{value}} {{units}}'
  },
  'pattern.velocityValueUnits': {
    'value': '|v<sub>{{index}}</sub>| = {{value}} {{units}}'
  },
  'pattern.unitsPower': {
    'value': '{{units}}<sup>{{power}}</sup>'
  },
  'a11y': {
    'restart': {
      'value': 'Restart'
    },
    'massSlider': {
      'value': 'Mass Slider'
    },
    'scaleSlider': {
      'value': 'Scale Slider'
    },
    'increase': {
      'value': 'Increase'
    },
    'decrease': {
      'value': 'Decrease'
    }
  }
};

function getRepeatedKeys( json1, json2, json3 ) {
  debugger;
  const keys1 = new Set( Object.keys( json1 ).concat( Object.keys( json1.a11y ) ) );
  const keys2 = new Set( Object.keys( json2 ).concat( Object.keys( json2.a11y ) ) );
  const keys3 = new Set( Object.keys( json3 ).concat( Object.keys( json3.a11y ) ) );
  const repeatedKeys = new Set();
  for ( const key of keys1 ) {
    if ( keys2.has( key ) || keys3.has( key ) ) {
      repeatedKeys.add( key );
    }
  }
  for ( const key of keys2 ) {
    if ( keys1.has( key ) || keys3.has( key ) ) {
      repeatedKeys.add( key );
    }
  }
  for ( const key of keys3 ) {
    if ( keys2.has( key ) || keys1.has( key ) ) {
      repeatedKeys.add( key );
    }
  }
  return Array.from( repeatedKeys );
}

console.log( 'not crazy' );
const repeatedKeys = getRepeatedKeys( mySolarStrings, keplersStrings, commonStrings );
console.log( repeatedKeys );






```
</details>

We also built my-solar-system and removed the unused `simDescription` key.

--------------------------


We are ready to proceed with Step 3 of https://github.com/phetsims/my-solar-system/issues/87#issuecomment-1449041208 when ready.

**Comment by zepumph** [comment]
Alright. Translations have been moved to their right spots. Closing
